source/fe/fe_hermite.cc

```diff
@@ -0,0 +1,840 @@
+
+template <int dim, int spacedim>
+std::vector<unsigned int>
+FE_Hermite<dim, spacedim>::get_lexicographic_to_hierarchic_numbering() const
+{
+  return internal::hermite_lexicographic_to_hierarchic_numbering<dim>(
+    this->regularity);
+}
+
+
+
+/*
+ * A special version of this function is needed for Hermite elements,
+ * due to the need to rescale different shape functions according to
+ * the derivative order they represent.
+ */
+template <int dim, int spacedim>
+void
+FE_Hermite<dim, spacedim>::fill_fe_values(
+  const typename Triangulation<dim, spacedim>::cell_iterator &,
+  const CellSimilarity::Similarity cell_similarity,
+  const Quadrature<dim> & /*quadrature*/,
+  const Mapping<dim, spacedim> &                           mapping,
+  const typename Mapping<dim, spacedim>::InternalDataBase &mapping_internal,
+  const dealii::internal::FEValuesImplementation::MappingRelatedData<dim,
+                                                                     spacedim>
+    & /*mapping_data*/,
+  const typename FiniteElement<dim, spacedim>::InternalDataBase &fe_internal,
+  dealii::internal::FEValuesImplementation::FiniteElementRelatedData<dim,
+                                                                     spacedim>
+    &output_data) const
+{
+  // Convert data object to internal data for this class.
+  // Fails with an exception if that is not possible.
+  Assert(
+    (dynamic_cast<const typename FE_Hermite<dim, spacedim>::InternalData *>(
+       &fe_internal) != nullptr),
+    ExcInternalError());
+  const typename FE_Hermite<dim, spacedim>::InternalData &fe_data =
+    static_cast<const typename FE_Hermite<dim, spacedim>::InternalData &>(
+      fe_internal);
+
+  Assert(
+    (dynamic_cast<const typename MappingHermite<dim, spacedim>::InternalData *>(
+       &mapping_internal) != nullptr) ||
+      (dynamic_cast<const typename MappingCartesian<dim>::InternalData *>(
+         &mapping_internal) != nullptr),
+    ExcInternalError()); /*
+const typename MappingHermite<dim, spacedim>::InternalData
+&mapping_internal_herm = static_cast<const typename MappingHermite<dim,
+spacedim>::InternalData &>( mapping_internal);*/
+
+  const UpdateFlags flags(fe_data.update_each);
+
+  // Transform values gradients and higher derivatives. Values also need to
+  // be rescaled according the the nodal derivative they correspond to.
+  if ((flags & update_values) &&
+      (cell_similarity != CellSimilarity::translation))
+    {
+      internal::Rescaler<dim, spacedim, double> shape_fix;
+      for (unsigned int i = 0; i < output_data.shape_values.size(0); ++i)
+        for (unsigned int q = 0; q < output_data.shape_values.size(1); ++q)
+          output_data.shape_values(i, q) = fe_data.shape_values(i, q);
+      shape_fix.rescale_fe_hermite_values(shape_fix,
+                                          *this,
+                                          mapping_internal, //_herm,
+                                          output_data.shape_values);

```

Comment:
Out of curiosity, couldn't one call `FE::Poly::fill_fe_values()` and to the rescaling as postprocessing? I know these lines of code from somewhere.